funnel/views/label.py

```diff
@@ -172,6 +172,22 @@ def archive(self):
             flash(_("CSRF token is missing"), category='error')
         return redirect(self.obj.project.url_for('labels'), code=303)
 
+    @route('unarchive', methods=['POST'])
+    @lastuser.requires_login
+    @requires_permission('admin')
+    def unarchive(self):
+        if not self.obj.archived:
+            flash(_("This label is not archived"), category='error')
+        else:
+            form = forms.Form()
+            if form.validate_on_submit():
+                self.obj.archived = False
+                db.session.commit()
+                flash(_("The label has been archived"), category='success')
+            else:
+                flash(_("CSRF token is missing"), category='error')
```

Comment:
Never tell a user "CSRF token is missing". What is the user supposed to do? There is no CSRF token in the UI. Instead, tell the user to submit again to confirm the action.